demos/link-prediction/homogeneous-comparison-link-prediction.ipynb

```diff
@@ -0,0 +1,1515 @@
+{
```

Comment:
<p>Maybe the construction of the <code>BiasedRandomWalk</code>  walker could be put into a function so that it doesn't have to be copied into each of the <code>_embedding</code>  functions, and the explanation of the <code>p</code>  parameter (and all the others for node2vec) can be placed in one place. For example, a section that has a title like "Node2Vec random walking" with the explanation of the parameters and a cell like:</p><p> </p><pre class="ql-syntax" spellcheck="false">def create_node2vec_walker():
    p = 1.0
    q = 1.0
    ...
    
    return BiasedRandomWalk(graph, n=num_walks, length=walk_length, p=p, q=q)
</pre><p> </p><p>Finally, I think with p=1 and q=1, "biased" random walk is equivalent to a uniform random walk. Did you choose those hyper parameters by just copying from https://stellargraph.readthedocs.io/en/latest/demos/link-prediction/node2vec-link-prediction.html ? (It's fine if so.)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1658/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
<p>I think it is fine to define a function to generate biased random walks, which can used by all embedding learning modules.</p><p> </p><p>For the setting of the value of p and q, I just follow the demo <a href="https://stellargraph.readthedocs.io/en/latest/demos/link-prediction/node2vec-link-prediction.html" target="_blank">https://stellargraph.readthedocs.io/en/latest/demos/link-prediction/node2vec-link-prediction.html</a>&nbsp;</p><p>In this demo notebook, we just give an example about how to use the validation set to choose the best link operator. It can also be used to choose the hyper parameters like p and q. Here, we only provide a default value for p and q, and leave room for uses to explore properer values.</p><p> </p>